201-vm-domain-join-existing/azuredeploy.json

```diff
@@ -0,0 +1,65 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "vm": {
+            "type": "string",
+            "metadata": {
+                "description": "Virtual machine to be domain joined."
+            }
+        },
+        "DomainJoinUserName": {
+            "type": "string",
+            "metadata": {
+                "description": "Domain NetBiosName plus User name of a domain user with sufficient rights to perfom domain join operation. E.g. domain\\username "
+            }
+        },
+        "DomainJoinUserPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Domain user password."
+            }
+        },
+        "domainFQDN": {
+            "type": "string",
+            "metadata": {
+                "description": "Domain FQDN where the virtual machine will be joined."
+            }
+        },
+        "ouPath": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "Specifies an organizational unit (OU) for the domain account. Enter the full distinguished name of the OU in quotation marks. Example: 'OU=testOU; DC=domain; DC=Domain; DC=com"
+            }
+        }
+    },
+    "variables": {
+        "domainJoinOptions": 3
+    },
+    "resources": [
+        {
+            "comments": "Join domain - JsonADDomainExtension",
+            "apiVersion": "2015-06-15",
+            "type": "Microsoft.Compute/virtualMachines/extensions",
+            "name": "[concat(parameters('vm'),'/joindomain')]",
+            "location": "[resourceGroup().location]",
```

Comment:
You're correct, must be at the same location of the VM, I'll make it a parameter.